blackmarket/Networking/Services/AuthenticationServices.swift

```diff
@@ -72,8 +72,19 @@ internal class AuthenticationServices {
             case .failure:
                 return .failure(AuthError.login)
             }
-        } catch {
-            return .failure(AuthError.request)
+        } catch let error {
+            guard
+                let responseError = error.asAFError,
```

Comment:
I don't think it is a good idea to couple to Alamofire here other than in the APIClient if using that network provider. PLease refactor this to receive an error from the APIClient that includes the statusCode